experiment_scheduler/master/master.py

```diff
@@ -52,15 +54,18 @@ def _execute_command(self, interval=1):
         :return: None
         """
         while True:
-            available_task_managers = self.get_available_task_managers()
-            for _ in self.queued_tasks:
-                if len(available_task_managers) > 0:
-                    task_manager = available_task_managers.pop(0)
-                    self.execute_task(task_manager)
-            # [TODO] master pipe recv logic required here
-            for task_manager, pipe in self.master_pipes.items():
-                if pipe.poll():
-                    print(pipe.recv())  # need change later
+            gpu_idx = responser.get_max_free_gpu()
+            if gpu_idx != -1:
+                available_task_managers = self.get_available_task_managers()
+                self.resource_monitor.queued_gpu_idx.append(gpu_idx)
```

Comment:
GPU id popped from queued_gpu_idx must ensure the gpu is in same available_task_manager.

Reply:
I recommend to group task_manger_address (from get_availabale_task_managers) and  gpu index into a tuple, returned from get_available_task_manager.